Library/ViewModels/CategorySelectionViewModel.swift

```diff
@@ -36,47 +41,48 @@ public final class CategorySelectionViewModel: CategorySelectionViewModelType,
 
     let orderedCategories = categoriesEvent.values().map(categoriesOrderedByPopularity)
 
-    self.loadCategorySections = orderedCategories.map { rootCategories in
-      var sectionTitles = [String]()
-      let categoriesData = rootCategories.compactMap { category -> [String]? in
-        guard let subcategories = category.subcategories?.nodes else {
-          return nil
-        }
-
-        let subcategoryNames = subcategories.map { $0.name }
-        sectionTitles.append(category.name)
-
-        return [Strings.All_category_name_Projects(category_name: category.name)] + subcategoryNames
-      }
-
-      return (sectionTitles, categoriesData)
-    }
-
-    let selectedCategoryIndexes = self.categorySelectedAtIndexPathProperty.signal.skipNil()
-      .scan(Set<IndexPath>.init()) { (selectedIndexes, currentIndexPath) -> Set<IndexPath> in
-        var updatedIndexes = selectedIndexes
+    self.loadCategorySections = orderedCategories.map(categoryData(from:))
 
-        if selectedIndexes.contains(currentIndexPath) {
-          updatedIndexes.remove(currentIndexPath)
-        } else {
-          updatedIndexes.insert(currentIndexPath)
-        }
+    let selectedCategoryIndexes = self.categorySelectedWithValueProperty.signal.skipNil()
+      .map(first)
+      .scan(Set<IndexPath>.init(), updatedSelectedValues(selectedValues:currentValue:))
 
-        return updatedIndexes
-      }
+    let selectedCategoryIds = self.categorySelectedWithValueProperty.signal.skipNil()
+      .map(second)
+      .scan(Set<Int>.init(), updatedSelectedValues(selectedValues:currentValue:))
 
     self.selectCellAtIndexProperty <~ selectedCategoryIndexes
       .takePairWhen(self.shouldSelectCellAtIndexProperty.signal.skipNil())
       .map { selectedCategoryIndexes, shouldSelectIndex in
         selectedCategoryIndexes.contains(shouldSelectIndex)
       }
 
-    self.goToCuratedProjects = self.continueButtonTappedProperty.signal.ignoreValues()
+    self.goToCuratedProjects = selectedCategoryIds
+      .takeWhen(self.continueButtonTappedProperty.signal)
+      .map { $0.sorted() }
+      .map(Array.init)
+
+    let selectedCategoriesCount = selectedCategoryIndexes.map { $0.count }
+
+    self.continueButtonEnabled = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(0),
```

Comment:
It actually needs to be an `Int` because it's merged with `selectedCategoriesCount`. Since `selectedCategoriesCount` only fires once a category is selected, we need to set the initial value to `0` on `viewDidLoad` so that the continue button becomes disabled when the view first loads.

Reply:
Hmm I see. So maybe mapping the `selectedCategoriesCount` instead would be more clear?
```
self.continueButtonEnabled = Signal.merge(
      self.viewDidLoadProperty.signal.mapConst(false),
      selectedCategoriesCount
        .map { $0 > 0 && $0 <= CategorySelectionViewModel.minimumCategorySelectionCount }
        .skipRepeats()
    )
```